parliament_challenge.py

```diff
@@ -0,0 +1,140 @@
+import requests
+import waitress
+import flask
+from flask import Flask, request, json
+
+
+def filter_speeches_dict(speeches):
+    """
+    Filtering speeches data for relevant keys.
+    Relevant keys are anforande_id, dok_datum,
+    parti, avsnittsrubrik, links and intressent_id.
+    """
+
+    speeches_list = []
+    for s in speeches:
+        speechesDict = {
+                "anforande_id": s.get("anforande_id"),
+                "dok_datum": s.get("dok_datum"),
+                "parti": s.get("parti"), 
+                "avsnittsrubrik": s.get("avsnittsrubrik"),
+                "links": [{ 
+                    "rel": "speech",
+                    "href": s.get("protokoll_url_www")
+                        }],
+                "intressent_id": s.get("intressent_id")
+                }
+        speeches_list.append(speechesDict)
+    return speeches_list
+
+
+def filter_member_dict(member):
+    """
+    Filtering member data for relevant keys.
+    Relevant keys are tilltalsnamn, valkrets,
+    bild_url_192, uppgift and intressent_id.
+    """
+
+    if member["personuppgift"] is None:
+        email = "null"
+    else:
+        for e in member["personuppgift"]["uppgift"]:
+            if e["kod"] == "Officiell e-postadress":
+                uppgift = e["uppgift"]
+                email = uppgift[0]
+    membersDict = {
+        "tilltalsnamn": member.get("tilltalsnamn"),
+        "valkrets": member.get("valkrets"),
+        "uppgift": email,
+        "bild_url_192": member.get("bild_url_192"),
+        "intressent_id": member.get("intressent_id")
+    }
+    return membersDict
+
+
+def get_member_data(intressent_id):
+    """
+    Get a member data from the member's api
+    and filter for relevant keys.
+    """
+
+    domain = 'http://data.riksdagen.se'
+    members = 'personlista'
+    format_type = 'json'
+    response = requests.get('{}/{}/?iid={}&utformat={}'.format(domain, members, intressent_id, format_type))
+    if response.status_code == 200:
+        data = response.json()
+        if data["personlista"]["@hits"] == "1":
```

Comment:
Add some line comments with the `#` here, it's good to comment what the logic is doing when it looks a bit complex.